recipes/gperftools/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class GperftoolsConan(ConanFile):
+    name = "gperftools"
+
+    description = "Google Performance Tools"
+    topics = ("conan", "gperftools", "tcmalloc", "cpu/heap profiler")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/gperftools/gperftools"
+    license = ["BSD-3-Clause License"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("gperftools is not supported by Visual Studio")
+        if self.settings.os == "Windows" and self.options.shared:
+            # libtool:   error: can't build i686-pc-mingw32 shared library unless -no-undefined is specified
+            raise ConanInvalidConfiguration("gperftools can't be built as shared on Windows")
```

Comment:
Did you try with CMakeLists? It could avoid msys2 build requirement (autotools are slow as hell on windows) and allow to build with Visual Studio.